cmd/usage-stats-handler/main.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+
+	"github.com/Jeffail/gabs"
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/cortexproject/cortex/pkg/util/flagext"
+	"github.com/go-kit/kit/log/level"
+	"github.com/grafana/usage-stats-handler/pkg/metricsutil"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/version"
+	"github.com/weaveworks/common/server"
+)
+
+func main() {
+	var (
+		serverConfig          server.Config
+		instrumentationConfig metricsutil.InstrumentationServerConfig
+		downstreamUrlFlag     string
+		applicationName       = "usage_stats_handler"
+	)
+
+	flagext.RegisterFlags(&serverConfig, &instrumentationConfig)
+	flag.StringVar(&downstreamUrlFlag, "usage-stats.downstream", "http://localhost:3000", "URL of the original usage stats handler")
+	flag.Parse()
+
+	util.InitLogger(&serverConfig)
```

Comment:
Is this going to cause problems when we switch to a version of cortex/pkg/util that includes cortexproject/cortex#3781 ?

Reply:
This is a larger issue with reaching into the cortex project and using components that they may not consider to be public APIs

Reply:
see also cortexproject/cortex#3779

Reply:
Good catch. Since we use weaveworks common in multiple projects we might have to address this in multiple places.